test/include/dlaf_test/matrix/util_generic_blas.h

```diff
@@ -0,0 +1,157 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2019, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#pragma once
+
+#include <functional>
+#include <tuple>
+
+#include <blas.hh>
+
+#include "dlaf_test/util_types.h"
+
+/// @file
+
+namespace dlaf {
+namespace matrix {
+namespace test {
+
+using namespace dlaf_test;
+
+/// Returns a tuple of element generators of three matrices A(m x m), B (m x n), X (m x n), for which it
+/// holds op(A) X = alpha B (n can be any value).
+///
+/// The tile elements of A matrix (el_op_a) are chosen such that:
+///   op(A)_ik = (i+1) / (k+.5) * exp(I*(2*i-k)) for the referenced elements
+///   op(A)_ik = -9.9 otherwise.
+///
+/// The X matrix elements (el_x) are computed as
+///   Xkj = (k+.5) / (j+2) * exp(I*(k+j)),
+/// where I = 0 for real types or I is the complex unit for complex types.
+/// These data are typically used to check whether the result of the equation
+/// performed with any algorithm is consistent with the computed values.
```

Comment:
`I` is already used in the definition of the elements of A. Please move this after line 33.